applications/finetune-quora-embeddings/optimize.py

```diff
@@ -0,0 +1,236 @@
+from modal import Stub, Image, gpu, Volume, NetworkFileSystem
+from optuna import Trial
+from helpers.data import format_dataset, score_prediction
+import torch.nn as nn
+from sklearn.metrics import accuracy_score, precision_score, recall_score, roc_auc_score
+import json
+import pandas as pd
+import math
+from typing import List
+
+# GPU Configuration
+gpu_config = gpu.A100()
+
+# Finetuning Configuration ( Arrays are configurable parameters )
+MODELS = [
+    # "BAAI/bge-base-en-v1.5",
+    "WhereIsAI/UAE-Large-V1"
+]
+DENSE_LAYER_DIMS = [2048]
+ACTIVATION_FUNCTIONS = ["Tanh"]  # This is a torch func, we use getattr to read it
+SCHEDULER = ["warmupconstant"]
+DATASET_SIZE = [128000]
+WARMUP_STEPS = [500]
+BATCH_SIZE = 64
```

Comment:
this hsould be a parameter

Reply:
32, 64, 128, etc

also no learning rate?

Reply:
I got an OOM error when working with 128 previously. I think 96 might be an option though.